### ISSUE TITLE
False positive MD057 on links in footnotes

### ISSUE THREAD

**Issue by martimlobao** [rvben/rumdl#286]
`test.md`
```md
# Title

A footnote[^1].

[^1]: [link](https://www.google.com).
```

```sh
❯ rumdl check test.md -e MD057
test.md:5:7: [MD057] Relative link '[link](https://www.google.com).' does not exist
```

**Comment by rvben** [comment]
Fixed in v0.0.215.

**Root cause:** Footnote definitions (`[^id]: content`) were being parsed as reference link definitions (`[label]: url`), causing MD057 to treat the footnote content as a relative URL and flag it as missing.

**Fix:** The reference definition parser now skips IDs starting with `^`, correctly excluding footnote definitions.

```markdown
[^1]: This footnote content is no longer flagged as a missing file
[ref]: ./actual-link.md  <!-- This still works as expected -->
```

Thanks for reporting!
